lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -230,394 +227,449 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
-  """
-  @spec card(keyword()) :: component()
+  @spec card(assigns :: Socket.assigns()) :: Phoenix.LiveView.Rendered.t()
   def card(assigns) do
-    assigns =
-      assigns
-      |> Map.new()
-      |> CardComponent.normalize_params()
-
-    {CardComponent, assigns}
+    ~H"""
+    <.card_title title={@title} hint={@hint} />
+    <div class={"banner-card mt-auto #{@class}"}>
+      <h6 class="banner-card-title" :if={@inner_title}>
+        <%= @inner_title %>
+        <.hint :if={@inner_hint} text={@inner_hint} />
+      </h6>
+      <div class="banner-card-value">
+        <%= render_slot(@inner_block) %>
+      </div>
+    </div>
+    """
   end
 
   @doc """
-  Renders a fields card component.
-
-  It can be rendered in any dashboard page via the `render_page/1` function:
-
-      def render_page(assigns) do
-        fields_card(
-          title: "Run queues",
-          inner_title: "Total",
-          fields: ["USER": "...", "ROOTDIR": "..."]
-        )
-      end
+  Fields card component.
 
   You can see it in use the Home page in the Environment section.
+  """
 
-  # Options
-
-  These are the options supported by the component:
-
-    * `:fields` - Required. A list of key-value elements that will be shown inside the card.
-
-    * `:title` - The title above the card.
-      Default: `nil`.
-
-    * `:inner_title` - The title inside the card.
-      Default: `nil`.
+  attr :fields, :list,
+    required: true,
+    doc: "A list of key-value elements that will be shown inside the card."
 
-    * `:hint` - A textual hint to show close to the title.
-      Default: `nil`.
+  attr :title, :string, default: nil, doc: "The title above the card."
+  attr :hint, :string, default: nil, doc: "A textual hint to show close to the title."
+  attr :inner_title, :string, default: nil, doc: "The title inside the card."
+  attr :inner_hint, :string, default: nil, doc: "A textual hint to show close to the inner title."
 
-    * `:inner_hint` - A textual hint to show close to the inner title.
-      Default: `nil`.
-  """
-  @spec fields_card(keyword()) :: component()
   def fields_card(assigns) do
-    assigns =
-      assigns
-      |> Map.new()
-      |> FieldsCardComponent.normalize_params()
-
-    {FieldsCardComponent, assigns}
+    ~H"""
+    <%= if @fields && not Enum.empty?(@fields) do %>
+      <.card_title title={@title} hint={@hint} />
+      <div class="fields-card">
+        <div class="card mb-4">
+          <div class="card-body rounded pt-3">
+            <h6 class="card-title" :if={@inner_title}>
+              <%= @inner_title %>
+              <.hint :if={@inner_hint} text={@inner_hint} />
+            </h6>
+            <dl :for={{k, v} <- @fields}>
+              <dt class="pb-1"><%= k %></dt>
+              <dd>
+                <textarea class="code-field text-monospace" readonly="readonly" rows="1"><%= v %></textarea>
+              </dd>
+            </dl>
+          </div>
+        </div>
+      </div>
+    <% end %>
+    """
```

Comment:
I don't like the concept to have so many "different card" components, we should try to create one customizable enough to support all our needs... Or maybe that's too ambicious :)

Reply:
The challenge is how to create our custom components without exposing CSS. So any example with CSS should be removed IMO.